pkg/core/logger/loggertest/logger.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package loggertest
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+)
+
+// New creates a testing logger that buffers the logs in memory and
+// logs in debug level. Check observer.ObservedLogs for more details.
+func New(name string) (*logger.Logger, *observer.ObservedLogs) {
+	core, obs := observer.New(zapcore.DebugLevel)
+
+	log := logp.NewLogger(
+		name,
+		zap.WrapCore(func(in zapcore.Core) zapcore.Core {
+			return zapcore.NewTee(in, core)
+		}))
+
+	return log, obs
+}
+
+// PrintObservedLogs consumes, formats and prints all log entries from obs,
+// one at a time, printFn. It calls `observer.ObservedLogs.TakeAll`,
+// therefore, after calling it, the ObservedLogs will be empty.
+func PrintObservedLogs(obs *observer.ObservedLogs, printFn func(a ...any)) {
+	rawLogs := obs.TakeAll()
```

Comment:
[suggestion/non-blocking] `TakeAll` modifies the state of `obs`, so ieally I would keep it out of this function. I understand that this makes the call to `PrintObservedLogs` a bit longer, user needs to pass `obs.TakeAll()` instead of just `obs` into the as the parameter. I'm leaving it up to you Anderson.